app/api/util/response.py

```diff
@@ -53,3 +75,20 @@ def success_response(
     status_code: int = 200,
 ) -> Response:
     return Response(status_code=status_code, message=message, data=data, warnings=warnings)
+
+
+def error_response(
+    status_code: HTTPException
+    | Type[HTTPException]
```

Comment:
So the various `Type[...]` are because you normally call this like `error_response(BadRequest, ..)` and are just passing in the class which each of these has the status code as a static value. But sometimes you might pass in an error that was actually thrown so you have an object, not the class itself. Both have the `status_code` value, so it's fine to use either.

I thought MyPy couldn't handle `Type[HTTPException]` for derived classes (maybe an older version couldn't?), but it looks like it's fine with that, so can simplify it.

Reply:
Ah I see. Thank you for explaining and simplifying.